lib/cdo/aws/s3.rb

```diff
@@ -76,7 +79,12 @@ def self.random
     end
 
     def self.public_url(bucket, filename)
-      Aws::S3::Object.new(bucket, filename, region: CDO.aws_region).public_url
+      Aws::S3::Object.new(
+        bucket,
+        filename,
+        client: create_client,
+        region: CDO.aws_region
```

Comment:
I think if you provide `:client` in the options hash here, you can remove `:region` - it's won't be used.

> **Options Hash (options):**
> - **`:client` (Client)** — When `:client` is not given, the options hash is used to construct a new Client object.
> 
> _[AWS SDK for Ruby V2 documentation](http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Object.html#initialize-instance_method)_

Maybe this was always redundant.  Do you think there's any reason we should pass this option in `self.connect_v2!` above?

Reply:
hm, good catch, perhaps this option has been redundant ever since we added the `AWS_DEFAULT_REGION` environment variable in #3995 (Sep 2015) - I'll avoid touching it for now (to avoid another hour-long full test-suite run on this PR), but perhaps it could be removed in the future.